cupy/cuda/runtime.pyx

```diff
@@ -306,6 +314,61 @@ cpdef deviceEnablePeerAccess(int peerDevice):
     check_status(status)
 
 
+###############################################################################
+# IPC operations
+###############################################################################
+cpdef ipcCloseMemHandle(intptr_t devPtr):
+    status = cudaIpcCloseMemHandle(<void*>devPtr)
+    check_status(status)
+
+
+cpdef ipcGetEventHandle(intptr_t event):
+    cdef driver.IpcEventHandle handle
+    status = cudaIpcGetEventHandle(&handle, <driver.Event>event)
+    check_status(status)
+    # We need to do this due to a bug in Cython that
+    # cuts out the 0 bytes in an array of chars when
+    # constructing the python object
+    # resulting in different sizes assignment errors
+    # when recreating the struct from the python 
+    # array of bytes
+    reserved = [<unsigned char>handle.reserved[i] for i in range(64)]
```

Comment:
Just curious, do you have a pointer to the Cython issue?

Reply:
Nope, I guess its because cython treats char* arrays as strings instead of buffers of bytes, so it automatically throws away all the data after the first 0 when converting it to a python object.

Reply:
You're right...Cython has a horribly long doc on string processing which mentioned this: 
> The above will therefore only work correctly for C strings that do not contain null bytes.

https://cython.readthedocs.io/en/latest/src/tutorial/strings.html#passing-byte-strings

Reply:
I discovered it the hard way 😇

Reply:
...and I would never google for it until someone told me this fact 😅

Reply:
By the way, despite a Python list (`reserved = [...]`) is used to hold the 64 bytes, it's actually as performant as a C++ vector. I was very surprised to learn it...

Reply:
Do you have some link to this? 🙂

Reply:
No, I tested it myself:
```cython
# distutils: language = c++

from libcpp cimport vector
from libc.string cimport memcpy

cdef int i
cdef char[64] data = [<char>i for i in range(64)]


cpdef f():
    cdef vector.vector[char] reserved
    cdef int i
    reserved.resize(64)

    memcpy(reserved.data(), data, 64)
    return bytes(reserved)


cpdef f2():
    cdef vector.vector[char] reserved = vector.vector[char](64)
    cdef int i

    memcpy(reserved.data(), data, 64)
    return bytes(reserved)


cpdef g():
    cdef list reserved = [<char>(data[i]) for i in range(64)]
    return bytes(reserved)
```
```python
In [1]: from test_vector_to_bytes import f, f2, g                                                                                                                      

In [2]: %timeit f()                                                                                                                                                    
831 ns ± 1.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: %timeit f2()                                                                                                                                                   
846 ns ± 0.859 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [4]: %timeit g()                                                                                                                                                    
817 ns ± 2.66 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [5]: f() == f2() == g()                                                                                                                                             
Out[5]: True
```